test/test_stateless.py

```diff
@@ -188,23 +221,41 @@ def test_tied_weights_warns(self):
                       'l1.bias': bias,
                       'buffer': buffer}
         x = torch.randn(1, 1)
-        self.assertNotWarn(lambda: stateless.functional_call(module, parameters, x))
+        self.assertNotWarn(lambda: stateless.functional_call(module, parameters, x, tie_weights=True))
 
         # if tied values are the same tensors, shouldn't warn
         parameters['tied_bias'] = bias
         parameters['tied_buffer'] = buffer
-        self.assertNotWarn(lambda: stateless.functional_call(module, parameters, x))
+        self.assertNotWarn(lambda: stateless.functional_call(module, parameters, x, tie_weights=True))
         del parameters['tied_bias']
         del parameters['tied_buffer']
 
-        with self.assertWarnsOnceRegex(UserWarning, "functional_call was passed multiple values"):
+        with self.assertRaisesRegex(ValueError, "functional_call got values for both (l1.bias|tied_bias)"):
             parameters['tied_bias'] = torch.tensor([5.0])
-            stateless.functional_call(module, parameters, x)
+            stateless.functional_call(module, parameters, x, tie_weights=True)
         del parameters['tied_bias']
 
-        with self.assertWarnsOnceRegex(UserWarning, "functional_call was passed multiple values"):
+        with self.assertRaisesRegex(ValueError, "functional_call got values for both (buffer|tied_buffer)"):
             parameters['tied_buffer'] = torch.tensor([5.0])
-            stateless.functional_call(module, parameters, x)
+            stateless.functional_call(module, parameters, x, tie_weights=True)
+
+
+    def test_tied_weights_no_error_without_kwarg(self):
```

Comment:
What does "without kwarg" mean here? Was this supposed to test what happens if we don't pass `tie_weights=False`?

Reply:
Changed name--it should have been with the false flag